bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/api/Authorization.java

```diff
@@ -0,0 +1,533 @@
+         * Accept-Language: de-DE,de;q=0.9
+         * Accept: text/html,application/xhtml+xml,application/xml;q=0.9,;q=0.8
+         *
+         *
+         * 2025-07-15 21:48:39.577 [WARN ] [rcedesme.internal.server.Authorization] - Failed request
+         * /as/authorization.oauth2client_id=62778dc4-1de3-44f4-af95-115f06a3a008&code_challenge_method=S256&redirect_uri=rismycar%3A%2F%2Flogin-callback&response_type=code&scope=email+profile+ciam-uid+phone+openid+offline_access&code_challenge=q-xTU0kQx3fLkhAO89qo_4shExS7wa6XVoG1DXAYoZ4
+         * - org.eclipse.jetty.http.BadMessageException: 500: Request header too large
+         **/
+        try {
+            loginHttpClient.start();
+            loginHttpClient.getProtocolHandlers().remove(WWWAuthenticationProtocolHandler.NAME);
+            String codeVerifier = generateCodeVerifier(32);
+            String codeChallenge = generateCodeChallengeSafely(codeVerifier);
+            String resumeUrl = getResumeUrl(loginHttpClient, codeChallenge);
+            sendUserAgent(loginHttpClient);
+            sendUsername(loginHttpClient);
+            String preLoginToken = performPasswordLogin(loginHttpClient);
+            String authCode = resumeAuthentication(loginHttpClient, resumeUrl, preLoginToken);
+            requestAccessToken(loginHttpClient, authCode, codeVerifier);
+        } catch (MercedesMeAuthException | MercedesMeApiException e) {
+            throw e;
+        } catch (Exception e) {
+            String message = e.getMessage();
+            throw new MercedesMeApiException(message == null ? "unknown" : message);
+        } finally {
+            try {
+                loginHttpClient.stop();
+            } catch (Exception e) {
+                logger.warn("Failed to stop HttpClient {}", e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Get URL for resume authentication.
+     *
+     * @throws MercedesMeAuthException if response status isn't correct or URL parameter not delivered
+     * @throws MercedesMeApiException if an error occurs during API call
+     */
+    private String getResumeUrl(HttpClient loginHttpClient, String codeChallenge)
+            throws MercedesMeAuthException, MercedesMeApiException {
+        Fields resumeContent = new Fields();
+        resumeContent.put("client_id", Constants.AUTH_CLIENT_ID);
+        resumeContent.put("code_challenge_method", "S256");
+        resumeContent.put("redirect_uri", Constants.AUTH_REDIRECT_URI);
+        resumeContent.put("response_type", "code");
+        resumeContent.put("scope", Constants.AUTH_SCOPE);
+        resumeContent.put("code_challenge", codeChallenge);
+
+        String resumeUrl = null;
+        Request resumeRequest = loginHttpClient
+                .newRequest(baseUrl + "/as/authorization.oauth2?" + FormContentProvider.convert(resumeContent))
+                .followRedirects(true);
+        resumeRequest.header(HttpHeader.USER_AGENT, Constants.AUTH_USER_AGENT);
```

Comment:
See comment further down.
```suggestion
        resumeRequest.agent(Constants.AUTH_USER_AGENT);
```